datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -344,12 +344,10 @@ impl CommonSubexprEliminate {
         self.try_unary_plan(expr, input, config)?
             .transform_data(|(mut new_expr, new_input)| {
                 assert_eq!(new_expr.len(), 1); // passed in vec![predicate]
-                let new_predicate = new_expr
-                    .pop()
-                    .unwrap()
-                    .unalias_nested()
-                    .update_data(|new_predicate| (new_predicate, new_input));
-                Ok(new_predicate)
+                let new_predicate = new_expr.pop().unwrap();
+                // the closure didn't transform the expr, but will be
```

Comment:
Is it possible to remove `transform_data`?
```rust
self.try_unary_plan(expr, input, config)?
            .map_data(|(mut new_expr, new_input)| {
                assert_eq!(new_expr.len(), 1); // passed in vec![predicate]
                let new_predicate = new_expr.pop().unwrap();
                Filter::try_new(new_predicate, Arc::new(new_input))
                    .map(LogicalPlan::Filter)
            })
```

Reply:
That is an excellent idea. It was -- done in 49d206e65